File: Iterations/code.java

import java.util.*;

class Solution {
    public int solution(int N) {
        // write your code in Java SE 8
        if (N <= 2)
            return 0;
        int leftPositionOfSetBit = -1, currPosition = 0, res = 0;
        while (N != 0) {
            int bit = N % 2;
            N = N / 2;
            if (bit == 1) {
                if (leftPositionOfSetBit == -1) {
                    leftPositionOfSetBit = currPosition;
                } else {
                    res = Math.max(res, currPosition - leftPositionOfSetBit - 1);
                    leftPositionOfSetBit = currPosition;
                }
                // System.out.print(currPosition+" ");
            }
            currPosition++;

        }
        return res;

    }
}

File: maxSilceSum/MaxSliceSum/code.js
function solution(A) {
    // write your code in JavaScript (Node.js 8.9.4)
    let N = A.length;
    let best = A[0], sum = 0;
    for (let x of A) {
        sum = Math.max(sum + x, x);
        best = Math.max(best, sum);
    }
    return best;
}